platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/components/wiring/ManualWiring.java

```diff
@@ -134,7 +133,7 @@ public AppCommunicationComponent wireAppCommunicationComponent(final Notificatio
             @NonNull final HaltRequestedConsumer haltRequestedConsumer,
             @NonNull final AppCommunicationComponent appCommunicationComponent,
             @NonNull final PreconsensusEventWriter preconsensusEventWriter,
-            @NonNull final Supplier<PlatformStatus> getPlatformStatus) {
+            @NonNull final PlatformStatusStateMachine platformStatusStateMachine) {
```

Comment:
I wonder if it might simplify dependencies if we created a simple functional interface like this:

```
interface CurrentPlatformStatus {
    PlatformStatus getCurrentStatus();
}
```

We can have the interface `PlatformStatusStateMachine` extend this interface. Then, for classes like this that just need the ability to poll the current status have a much simpler API to interact with.

Reply:
Done